sources/packages/backend/apps/api/src/route-controllers/student/student.controller.service.ts

```diff
@@ -133,37 +168,34 @@ export class StudentControllerService {
       `attachment; filename=${studentFile.fileName}`,
     );
 
-    // Temporary code to be changed in the upcoming effort
-    // when the the files will no longer be saved on DB.
-    const stopwatchLabel = `Downloaded file ${uniqueFileName}`;
-    console.time(stopwatchLabel);
     try {
+      this.logger.log(
+        `Downloading the file ${studentFile.fileName} from S3 storage.`,
+      );
       const fileContent = await this.objectStorageService.getObject(
         uniqueFileName,
       );
-      console.info("Downloaded file using S3 storage.");
+      this.logger.log(
+        `Downloaded the file ${studentFile.fileName} from S3 storage.`,
+      );
       // Populate file information received from S3 storage.
       response.setHeader("Content-Type", fileContent.contentType);
       response.setHeader("Content-Length", fileContent.contentLength);
+      // The readable stream is piped to the response as and when the
+      // the bytes are received so as to avoid it from getting accumulated
+      // in the memory (flow mode). This is in contrast to writing the
+      // bytes to the response collectively after accumulating them in memory
+      // as would be the case if reading the stream in the paused mode
+      // using the read method.
       fileContent.body.pipe(response);
     } catch (error: unknown) {
       if (error instanceof NoSuchKey) {
-        console.info("File not present on S3 storage.");
+        this.logger.log(
+          `File ${studentFile.fileName} is not present on S3 storage.`,
+        );
       } else {
-        console.error(parseJSONError(error));
+        this.logger.error(parseJSONError(error));
```

Comment:
We should throw an error from here, otherwise, the API will return a successful result.
```ts
throw new InternalServerErrorException(
  "Error while downloading the file.",
);
```